src/ChatPage.cpp

```diff
@@ -316,6 +316,65 @@ ChatPage::bootstrap(QString userid, QString homeserver, QString token)
     try {
         cache::init(userid);
 
+        connect(cache::client(), &Cache::databaseReady, this, [this]() {
+            nhlog::db()->info("database ready");
+
+            const bool isInitialized = cache::isInitialized();
+            const auto cacheVersion  = cache::formatVersion();
+
+            try {
+                if (!isInitialized) {
+                    cache::setCurrentFormat();
+                } else {
+                    if (cacheVersion == cache::CacheVersion::Current) {
+                        loadStateFromCache();
+                        return;
+                    } else if (cacheVersion == cache::CacheVersion::Older) {
+                        if (!cache::runMigrations()) {
+                            QMessageBox::critical(
+                              this,
+                              tr("Cache migration failed!"),
+                              tr("Migrating the cache to the current version failed. "
+                                 "This can have different reasons. Please open an "
+                                 "issue and try to use an older version in the mean "
+                                 "time. Alternatively you can try deleting the cache "
+                                 "manually."));
+                            QCoreApplication::quit();
+                        }
+                        loadStateFromCache();
+                        return;
+                    } else if (cacheVersion == cache::CacheVersion::Newer) {
+                        QMessageBox::critical(
+                          this,
+                          tr("Incompatible cache version"),
+                          tr("The cache on your disk is newer than this version of Nheko "
+                             "supports. Please update or clear your cache."));
```

Comment:
should probably change to 'Please update *nheko'

Reply:
Hey, that was there for a year or so already! But fine ;p